### PR TITLE
[t127823] Make sure you only access with sudo to the ir.model

### PR DESCRIPTION
**Pull request by ghost**
<div> Links to Odoo: </div> <ul>
<li><a target="_blank" href="https://odoo.bt-group.com/web#view_type=form&model=project.task&id=127823">[t127823] non variant forming attributes on product.product</a></li>
</ul>
<div>Affected Modules:</div>
<table><thead><tr><th>Module</th><th>Ext</th></tr></thead>
<tr><td>product_attribute_set</td><td>.py, .xml</td></tr>
<tr><td>attribute_set</td><td>.csv</td></tr>
</tbody></table>

### CODE REVIEW

File: attribute_set/models/attribute_set_owner.py [hunk 1]
```diff
@@ -33,18 +33,22 @@ def _build_attribute_eview(self):
...
     @api.model
     def remove_native_fields(self, eview):
         """Remove native fields related to native attributes from eview"""
-        native_attrs = self.env["attribute.attribute"].search(
-            [
-                ("model_id.model", "=", self._name),
-                ("attribute_set_ids", "!=", False),
-                ("nature", "=", "native"),
-            ]
+        native_attrs = (
```
Comment (BT-aestebanez) [inline attribute_set/models/attribute_set_owner.py]: why the parenthesis?
why no simply add sudo()?

**Review by BT-aestebanez — COMMENTED** [review]
I think the repo has more accesses to ir_model without sudo for example, this line:
relation_model_obj = self.env[self.relation_model_id.model]

File: attribute_set/models/attribute_set_owner.py (hunk 1 above)
Comment (ghost) [reply attribute_set/models/attribute_set_owner.py]: Hi, I got this suggestion in the pre-commit check: https://github.com/brain-tec/odoo-pim/runs/5315524570?check_suite_focus=true

Comment (BT-aestebanez) [reply attribute_set/models/attribute_set_owner.py]: thanks for the clarification :)

**Review by BT-mgomez — APPROVED** [review]

**Review by BT-aestebanez — COMMENTED** [review]
LGTM

**Pull request merged**